cedar-lean/Cedar/Partial/Entities.lean

```diff
@@ -0,0 +1,90 @@
+/-
+ Copyright 2022-2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-/
+
+import Cedar.Spec.Entities
+import Cedar.Partial.Value
+
+/-!
+This file defines Cedar partial-entities structures.
+-/
+
+namespace Cedar.Partial
+
+open Cedar.Data
+open Cedar.Spec (Attr EntityUID Result)
+
+/--
+Represents the information about one entity.
+
+Currently, this allows attrs to be known-to-exist-but-unknown-value,
+but does not allow attrs to be unknown-whether-they-exist-entirely.
+(the result of `e has attr` is never a residual for an `e` that is known to exist.)
+
+Currently, this does not allow any unknowns about ancestor information.
+All ancestor information must be fully concrete.
+-/
```

Comment:
Is this also the current state of the Rust implementation?

Reply:
I believe so, but @aaronjeline is the authority on that